composer.json

```diff
@@ -121,7 +121,10 @@
     "scripts": {
         "phing-install-dependencies": ["phing patch-dependencies", "phing installsolr installswaggerui"],
         "post-install-cmd": "@phing-install-dependencies",
-        "post-update-cmd": "@phing-install-dependencies",
+        "post-update-cmd": [
+            "@phing-install-dependencies",
+            "[ $COMPOSER_DEV_MODE -eq 0 ] || (npm install && phing copynodemodules)"
```

Comment:
The idea I had was that updating the dependencies would be a dev operation since the node modules are not needed for normal operation but are only used for updating the files in theme's js/vendor. This also means that files tracked by git could change by the update operation. But this was just an idea, and doing it regardless of mode would probably work fine since at least as long as versions are pinned.